### PR TITLE
Dom not found error to explain that jsdom needs to be initialised before using mount

### PR DESCRIPTION
**Pull request by ldabiralai**
This solves #223 

Not sure if we want to be linking to the docs from the code though?

What are peoples thoughts on this?

### CODE REVIEW

**Comment by lelandrichardson** [comment]
I agree this message should be changed, but we probably don't need to mention jsdom. Mount will work fine if it has a global document, which could be provided by a browser or a library like jsdom.

Maybe the message should be:
```
\`mount()\` can only be used when a global `document` exists.
```

**Comment by ljharb** [comment]
(due to a linter fix which only permits backticks when there's interpolation, this will need to be rebased)

File: src/ReactWrapper.js [hunk 1]
```diff
@@ -61,8 +61,8 @@ export default class ReactWrapper {
   constructor(nodes, root, options = {}) {
     if (!global.window && !global.document) {
       throw new Error(
-        'It looks like you called `mount()` without a jsdom document being loaded. ' +
-        'Make sure to only use `mount()` inside of a `describeWithDOM(...)` call. '
+        'It looks like you called \`mount()\` without a jsdom document being loaded. ' +
+        'Make sure to only use \`mount()\` after initialising jsdom (see http://airbnb.io/enzyme/docs/guides/jsdom.html)'
```
Comment (ljharb) [inline src/ReactWrapper.js]: no need to escape the backticks in a single-quoted string

File: src/ReactWrapper.js [hunk 2]
```diff
@@ -62,7 +62,7 @@ export default class ReactWrapper {
     if (!global.window && !global.document) {
       throw new Error(
         'It looks like you called `mount()` without a jsdom document being loaded. ' +
-        'Make sure to only use `mount()` inside of a `describeWithDOM(...)` call. '
+        'Make sure to only use `mount()` after initialising jsdom (see http://airbnb.io/enzyme/docs/guides/jsdom.html)'
```
Comment (ljharb) [inline src/ReactWrapper.js]: i agree with @lelandrichardson here - enzyme works in browsers too, so it's not about jsdom as it is about having a global `document`.

**Comment by lelandrichardson** [comment]
@ldabiralai do you mind squashing your commits into one commit and rebasing?

**Comment by ldabiralai** [comment]
Thanks for the feedback both, I've made the changes and squashed the commits into one.

**Comment by lelandrichardson** [comment]
:+1:

**Pull request merged**